services/catarse/spec/actions/billing/payments/expire_overdue_payments_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Billing::Payments::ExpireOverduePayments, type: :action do
+  describe 'Inputs' do
+    subject { described_class.inputs }
+
+    it { is_expected.to be_empty }
+  end
+
+  describe 'Outputs' do
+    subject { described_class.outputs }
+
+    it { is_expected.to be_empty }
+  end
+
+  describe '#call' do
+    subject(:result) { described_class.result }
+
+    let(:scope) do
+      Billing::Payment.where(id: create_list(:billing_payment, 2, :with_pix, :waiting_payment).pluck(:id))
+    end
+    let(:payment_a) { scope.first }
+    let(:payment_b) { scope.last }
+
+    before do
+      allow(Billing::Payment).to receive(:can_be_expired).and_return(scope)
+      allow(scope).to receive(:find_each).and_yield(payment_b).and_yield(payment_a)
+    end
+
+    it { is_expected.to be_success }
+
+    it 'expires overdue payments' do
+      expect([payment_a, payment_b]).to all(receive(:expire!))
+
+      result
+    end
+
+    context 'when payment expiration raise error' do
+      let(:error) do
+        Sentry.capture_message(
+          'Payment cannot transition to overdue',
+          level: :fatal,
+          extra: { payment_id: payment_a.id }
+        )
+      end
```

Comment:
**Problema**
Tu não pode colocar esse Sentry.capture_message aqui porque tu vai estar de fato capturando uma mensagem no momento que tu usar a variável `error` e não quando o erro for lançado.
A lógica nesse caso é a seguinte:
Criar um erro qualquer para ser o erro a ser lançado.
```ruby
let(:error) { StandardError.new('some error') }
```
O `allow(payment_a).to receive(:expire!).and_raise(error)` está correto.
Já dentro do teste, ficaria assim:
```ruby
it 'handles error with Sentry' do
  expect(Sentry).to receive(:capture_exception).with(colocar_os_params_aqui)
  
  result
end
```